src/apps/profiles/views.py

```diff
@@ -18,6 +25,37 @@ class LogoutView(auth_views.LogoutView):
     pass
 
 
+class UserDetail(LoginRequiredMixin, DetailView):
+    queryset = User.objects.all()
+    template_name = 'profiles/detail.html'
+    slug_url_kwarg = 'username'
+    query_pk_and_slug = True
+
+    def get_object(self, *args, **kwargs):
+        user = super().get_object(*args, **kwargs)
+        authorized = self.request.user.is_superuser or self.request.user == user
+        if authorized:
+            return user
+        raise Http404()
+
+    def get_context_data(self, **kwargs):
```

Comment:
@ckcollab What do you think of this server-side context pass? IMO, it'd be simpler, or at least conventional, to make that one request to the API that already exists when the profile edit/detail pages load. With all that said, this is pretty crafty and I don't think I've seen it done this way in our codebases, so props for that @Logan-Ruf.

Reply:
Thanks! 😃 Got the idea from some latency on the public competition page. Since detail views query for the object anyway, I figure it would be faster to send that with the original load. Plus, if a user has high latency, they have to wait for the page to finish loading before it queries the API for data, then they have to wait for the data from API. In this case, it is quite a bit faster. Plus, the UX is a lot smoother when everything loads at once. It feels snappy.

Reply:
Bump @ckcollab